### ISSUE TITLE
[RMP] Create a standard set of cross-framework evaluation metrics

### ISSUE THREAD

**Issue by benfred** [NVIDIA-Merlin/models#450]
# Problem statement
Merlin models should allow you to train models from different frameworks (TF/PT/lightfm/xgboost/implicit etc) and then compare them in a consistent "apples-to-apples" type way. Right now each framework has its own evaluation metrics, that aren't directly comparable to other frameworks. During the paper experiments for instance, we had to implement custom metric code for implicit to match the output from TF.  Customers also want to be able to compare to their own metrics.

# Goals
- Customers are able to compare models across frameworks.  
- Coverage of metrics: Precision, NDCG, MAP, AUC
- Support for both retrieval and ranking models
- Only used during final evaluation process.  This isn't used during training. (Non goal) 
- Example 

# Constraints
- Scoring should happen for all items, not just negative samples, in order to be able to compare correctly for ranking metric
- Only one implementation of these metrics so that the calculations are consistent.  This implies conversion.
- We want to be able to compare across multiple frameworks.

# Notes 
- Gabriel has a POC of cupy code: https://github.com/rapidsai/recsys/tree/main/benchmark_recsys/code/evaluation 
- Marc built a system like this at Spotify which converted to specific framework metrics

# Tasks
## 22.07 ( Ground work for cross framework evaluation )

- [ ] Create a mechanism for transferring data across frameworks
  - [ ] https://github.com/NVIDIA-Merlin/core/pull/102 

## 22.08 ( Enter the goal here )

- [ ] Establish a standard set of metrics (to be used in conjunction with cross-framework data transfer)
- [ ] Create example for cross framework evaluation metrics

**Comment by jperez999** [comment]
Constraints/Goals Add-On
- As a user I want to be able to use the data from my model prediction to another frameworks metrics to check if how my metrics compare to those of supported frameworks. i.e. XGBoost predictions going through tensorflow metric. 
- As a user I would like to be able to create metrics easily requiring a simple api like that of an operator. 
- As a user I should be able to create operators that can work in both GPU and non-GPU environments.

**Comment by viswa-nvidia** [comment]
@karlhigley  , @jperez999 , we discussed that we are doing the ground work for this in 22.07. Could you please add the tasks that we will complete in 22.07 above in 22.07 and the rest of it under 22.08. If you think this is going to go beyond 22.08, please put it in the future section and we can discuss this again in the grooming meeting. Thanks

**Comment by viswa-nvidia** [comment]
@karlhigley , please complete your scope discussion with Even and update this ticket and move it back to planned once the scope is defined.
